### PR TITLE
Refactor ApplicationBaseComponent

### PR DESCRIPTION
**Pull request by irfanhabib**
ApplicationBaseComponent is a thin wrapper component that initialises the EntityService and ApplicationService.

All tabs are nested now under ApplicationTabsComponent.

### CODE REVIEW

File: src/app/features/applications/application/application-base.component.ts [hunk 1]
```diff
@@ -69,159 +62,11 @@ const entityServiceFactory = (
 export class ApplicationBaseComponent implements OnInit, OnDestroy {
```
Comment (richard-cox) [inline src/app/features/applications/application/application-base.component.ts]: The OnInit and OnDestroy functions (and implements) lines can be removed if we're not using them. However I think with the rework below we might still do

File: src/app/features/applications/application/application-tabs-base/application-tabs-base.component.ts [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+        this.appDefaultEdits = {
+          name: application.app.entity.name,
+          instances: application.app.entity.instances,
+          memory: application.app.entity.memory,
+          enable_ssh: application.app.entity.enable_ssh,
+          environment_json: application.app.entity.environment_json
+        };
+      }));
+
+    const appSub = this.applicationService.app$.subscribe(app => {
```
Comment (richard-cox) [inline src/app/features/applications/application/application-tabs-base/application-tabs-base.component.ts]: We should move this into the application base

**Comment by irfanhabib** [comment]
@richard-cox I've moved both polling related observables to the `application-base`.

File: src/app/features/applications/application/application-base.component.ts [hunk 3]
```diff
@@ -61,12 +64,36 @@ const entityServiceFactory = (
...
+    private store: Store<AppState>,
+    private entityService: EntityService
+
   ) { }
 
-  ngOnDestroy(): void {
-  }
   ngOnInit(): void {
+
+    const { cfGuid, appGuid } = this.applicationService;
```
Comment (richard-cox) [inline src/app/features/applications/application/application-base.component.ts]: If we've moved this over from app base we should nuke if from there as well

File: src/app/features/applications/application/application-base.component.ts [hunk 4]
```diff
@@ -61,12 +64,36 @@ const entityServiceFactory = (
...
+    // Auto refresh
+    this.entityServiceAppRefresh$ = this.entityService.poll(10000, this.autoRefreshString).do(() => {
```
Comment (richard-cox) [inline src/app/features/applications/application/application-base.component.ts]: If we've moved this over from app base we should nuke if from there as well

**Pull request merged**